### PR TITLE
chore(cli): add --clean option to build command

### PR DESCRIPTION
**Pull request by shulaoda**
## Related Links
Related to #46 

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/cli/commands.ts [hunk 1]
```diff
@@ -41,10 +45,21 @@ export function runCli(): void {
...
   buildCommand
     .option('-w --watch', 'turn on watch mode, watch for changes and rebuild')
+    .option('--clean', 'clean the output directory before building')
     .description('build the library for production')
     .action(async (options: BuildOptions) => {
       try {
         const rslibConfig = await loadConfig(options.config, options.envMode);
+
+        if (options.clean) {
+          const distDir = rslibConfig.output?.distPath?.root ?? OUTPUT_DIST_DIR;
```
Comment (Timeless0911) [inline packages/core/src/cli/commands.ts:55]: It's raw Rsbuild config, we should get `distDir` from final config.

Comment (Timeless0911) [reply packages/core/src/cli/commands.ts:55]: Rsbuild already support clean [output.cleanDistPath](https://rsbuild.dev/zh/config/output/clean-dist-path#outputcleandistpath), may be you should implemeny clean feature in plugin-dts, to add an option of clean, and in default rslib core logic, when we use builtin plugin-dts, pass `clean: true` to it.

Comment (Timeless0911) [reply packages/core/src/cli/commands.ts:55]: The issue was written early and may have some problems now.

**Comment by Timeless0911** [comment]
I have changed the feature issue 
image
 
What do you think about the clean feature?

File: packages/core/src/cli/commands.ts (hunk 1 above)
Comment (shulaoda) [reply packages/core/src/cli/commands.ts:55]: Would it be confusing to include the `clean` option in `plugin-dts`?

Comment (shulaoda) [reply packages/core/src/cli/commands.ts:55]: Because `clean` is a global build option, not `plugin-dts`.

Comment (Timeless0911) [reply packages/core/src/cli/commands.ts:55]: We hope `rsbuild-plugin-dts` can be used seprately in a Rsbuild project, maybe we can inherit `output.cleanDistPath` options of Rsbuild and use its value to do DTS output clean?

Comment (shulaoda) [reply packages/core/src/cli/commands.ts:55]: Maybe I misunderstood. Do we need to clear `DTS` output instead of `dist` output before building?

Comment (Timeless0911) [reply packages/core/src/cli/commands.ts:55]: Js output will be cleaned according to Rsbuild `output.cleanDistPath` by default, the DTS output should align with this behaviour. Maybe we don't need a cli option to do that, just in built-in logic.

Comment (shulaoda) [reply packages/core/src/cli/commands.ts:55]: > Maybe we don't need a cli option to do that, just in built-in logic.

Agree

Comment (shulaoda) [reply packages/core/src/cli/commands.ts:55]: However, `Rsbuild` seems to default to clear `dist`. What is the significance of doing so? Does `plugin-dts` not support default clear?

Comment (Timeless0911) [reply packages/core/src/cli/commands.ts:55]: The `dist` maybe dirty in local development, clean behavihour can ensure the new output will not be influenced by existed output.

Comment (Timeless0911) [reply packages/core/src/cli/commands.ts:55]: `plugin-dts` have not supported it yet.

We should clean:
- bundleless mode: the tsc emit path
- bundle mode: the tsc emit temp path `.rslib/declarations` and api-extractor emit path

**Comment by shulaoda** [comment]
I tried to rewrite, but it seems that an empty commit caused PR to close.

**Comment by fi3ework** [comment]
Feel free to re-create one and link to this PR.

**Pull request closed without merging**